### PR TITLE
[DO NOT MERGE] Source Projects, zoom to layer extent. WinDrive fix

### PR DESCRIPTION
**Pull request by wilhelmberg**
- app: fix clamp function to always stay within WebMerc extent
- app/server: use omnivore for file based data and mapnik for others

Waiting for @amyleew 's feedback. :white_check_mark: 
refs https://github.com/mapbox/mapbox-studio-classic/issues/1487

**EDIT**
Also includes `middleware` fix for projects being saved to the wrong drive on Windows, e.g. drive `D:` was selected, but project was nevertheless still saved to `C:`.
refs https://github.com/mapbox/mapbox-studio-classic/issues/1485

### CODE REVIEW

File: index-shell.js [hunk 1]
```diff
@@ -46,7 +46,11 @@ var shellLog = path.join(process.env.HOME, '.mapbox-studio', 'shell.log');
 log(shellLog, 10e6, shellsetup);
 
 function shellsetup(err){
-
+    if(err){
```
Comment (flippmoke) [inline index-shell.js]: Please add spaces

File: lib/log.js [hunk 2]
```diff
@@ -21,13 +22,25 @@ module.exports = function(filepath, maxsize, callback) {
 
 
     function setup(offset) {
+        var dir = path.dirname(filepath);
+        if(!fs.existsSync(dir)){
```
Comment (flippmoke) [inline lib/log.js]: Same as above, please add spaces

File: lib/middleware.js [hunk 3]
```diff
@@ -21,10 +21,13 @@ middleware.normalizePaths = function(req, res, next) {
         //sometimes there are forward slashes and
         //sometimes there are backword slashes
         //doesn't seem to hapen with sources
-        //var is_windows = (req.query[key].indexOf(':\\') >= 0 || req.query[key].indexOf(':/') >= 0);
-        var is_windows = (req.query[key].indexOf(':\\') >= 0);
+        var is_windows = ('win32' === process.platform);
```
Comment (flippmoke) [inline lib/middleware.js:24]: Does this work for 64 bit windows?

Comment (wilhelmberg) [inline lib/middleware.js:24]: Yes! It's always `win32`.

```
node -e "console.log(process.arch, process.platform);"
```

prints  

```
x64 win32
```

**Pull request closed without merging**